app/frontend/components/dictionary_info_box.js

```diff
@@ -313,35 +311,27 @@ class DictionaryInfoBox {
 
     var extraDictContent = await this.getExtraDictionaryContent(lang, strongsEntry);
 
-    extendedStrongsInfo += "<b>" + strongsShortInfo + "</b>";
-    extendedStrongsInfo += "<p>";
-    extendedStrongsInfo += findAllLink;
-    extendedStrongsInfo += " | ";
-    extendedStrongsInfo += blueLetterLink;
-    extendedStrongsInfo += "</p>";
-    extendedStrongsInfo += extraDictContent;
-    extendedStrongsInfo += "<b>Strong's</b>";
-    extendedStrongsInfo += "<pre class='strongs-definition'>";
-    extendedStrongsInfo += strongsEntry.definition;
-    extendedStrongsInfo += "</pre>";
-
-    if (strongsEntry.references.length > 0) {
-      extendedStrongsInfo += "<hr></hr>";
-      var relatedStrongs = "<b>" + i18n.t("dictionary-info-box.related-strongs") + ":</b><br/>";
-      extendedStrongsInfo += relatedStrongs;
-      extendedStrongsInfo += "<table class='strongs-refs'>";
-
-      for (var i = 0;  i < strongsEntry.references.length; i++) {
-        var isLastRow = (i == (strongsEntry.references.length - 1));
-        var referenceTableRow = await this.getStrongsReferenceTableRow(strongsEntry.references[i], isLastRow);
-
-        if (referenceTableRow != null) {
-          extendedStrongsInfo += referenceTableRow;
-        }
-      }
+    var relatedStrongs = (await Promise.all(strongsEntry.references.map(async (ref, i) => 
+      await this.getStrongsReferenceTableRow(ref, i == (strongsEntry.references.length - 1))
+    ))).join('');
```

Comment:
While this is a nice optimization, it's not so easy to ready any more. Would you mind adding a comment above this statement, that explains what's happening here?

It maybe even worth a separate function - then the function name would explain what is happening.